store/localcachelayer/channel_layer.go

```diff
@@ -63,3 +81,79 @@ func (s LocalCacheChannelStore) GetMemberCountFromCache(channelId string) int64
 
 	return count
 }
+
+// ChannelCacheByName methods
+func (s LocalCacheChannelStore) GetByName(teamId string, name string, allowFromCache bool) (*model.Channel, *model.AppError) {
+	return s.getByName(teamId, name, false, allowFromCache)
+}
+func (s LocalCacheChannelStore) GetByNames(teamId string, names []string, allowFromCache bool) ([]*model.Channel, *model.AppError) {
+	var channels []*model.Channel
+
+	if allowFromCache {
+		var misses []string
+		visited := make(map[string]struct{})
+		for _, name := range names {
+			if _, ok := visited[name]; ok {
+				continue
+			}
+			visited[name] = struct{}{}
+			if cacheItem, ok := s.rootStore.channelByNameCache.Get(teamId + name); ok {
+				if s.rootStore.metrics != nil {
+					s.rootStore.metrics.IncrementMemCacheHitCounter("Channel By Name")
+				}
+				channels = append(channels, cacheItem.(*model.Channel))
+			} else {
+				if s.rootStore.metrics != nil {
+					s.rootStore.metrics.IncrementMemCacheMissCounter("Channel By Name")
+				}
+				misses = append(misses, name)
+			}
+		}
+		names = misses
+	}
+
+	if len(names) > 0 {
+		props := map[string]interface{}{}
+		var namePlaceholders []string
+		for _, name := range names {
+			key := fmt.Sprintf("Name%v", len(namePlaceholders))
+			props[key] = name
+			namePlaceholders = append(namePlaceholders, ":"+key)
+		}
+
+		dbChannels, err := s.ChannelStore.GetByNames(teamId, names, allowFromCache)
+
+		if err == nil {
+			return channels, err
+		}
+
+		for _, channel := range dbChannels {
+			s.rootStore.doStandardAddToCache(s.rootStore.channelByNameCache, teamId+channel.Name, channel)
+			channels = append(channels, channel) // add missing channels to the ones just found
+		}
+	}
+
+	return channels, nil
+}
+
+func (s LocalCacheChannelStore) getByName(teamId string, name string, includeDeleted bool, allowFromCache bool) (*model.Channel, *model.AppError) {
+	if allowFromCache {
+		if cacheItem, ok := s.rootStore.channelByNameCache.Get(teamId + name); ok {
```

Comment:
If you use here the `s.rootStore.doStandardReadCache` method, you can remove the Increment/Decrement counters because is already done there.